lib/iso_spinner.py

```diff
@@ -0,0 +1,158 @@
+        try:
+            with open(comps_xml_file, 'wt') as f:
+                f.write(comps_xml_str)
+        except IOError:
+            LOG.error("Failed to write XML to %s file." % comps_xml_file)
+            raise
+
+        comps_xml_chroot_path = os.path.join("/", comps_xml_file)
+        self._run_mock_command("--copyin %s %s" %
+                               (comps_xml_file, comps_xml_chroot_path))
+
+        LOG.debug("Creating spin repo")
+        createrepo_cmd = "createrepo -v -g %s %s" % (comps_xml_chroot_path,
+                                                     mock_spin_repo_dir)
+        self._run_mock_command("--shell '%s'" % createrepo_cmd)
+
+    def _create_spin_kickstart(self):
+        kickstart_file = self.config.get('kickstart_file')
+        LOG.info("Creating spin kickstart file %s" % kickstart_file)
+
+        with open(kickstart_file, "wt") as f:
+            repo_urls = self.config.get('distro_repo_url')
+            mock_spin_repo_name = self.config.get('mock_spin_repo').get('name')
+            mock_spin_repo_dir = self.config.get('mock_spin_repo').get('dir')
+            repo_urls[mock_spin_repo_name] = "file://%s/" % mock_spin_repo_dir
+            # FIXME: linux-firmware provided by HostOS conflicts with
+            #        ivtv-firmware, this hack must be removed as soon as its
+            #        problem has been sorted out
+            exclude_packages = "--excludepkgs=ivtv-firmware"
+            for name, url in repo_urls.items():
+                repo = ("repo --name=%s --baseurl=%s %s\n" %
+                        (name, url, exclude_packages))
+                f.write(repo)
+
+            f.write("%packages\n")
+            package_group_list = self.config.get('package_group_list')
+            hostos_package_group_name = (
+                "@%s" % self.config.get('hostos_package_group').get('name'))
+            if hostos_package_group_name not in package_group_list:
+                package_group_list.append(hostos_package_group_name)
+            for package_group in package_group_list:
+                f.write(package_group + "\n")
+
+            # FIXME: linux-firmware provided by HostOS conflicts with
+            #        ivtv-firmware, this hack must be removed as soon as its
+            #        problem has been sorted out
+            f.write("-ivtv-firmware\n")
+
+            f.write("%end\n")
+
+        self._run_mock_command("--copyin %s /" % kickstart_file)
+
+    def _spin(self):
+        LOG.info("Spinning ISO")
+        spin_cmd = ("pungi -c %s --nosource --nodebuginfo --name %s --ver %s" %
+                    (self.config.get('kickstart_file'),
+                     self.distro, self.version))
+        self._run_mock_command("--shell '%s'" % spin_cmd)
+
+    def _save(self):
+        iso_file = "%s-DVD-%s-%s.iso" % (self.distro, self.arch, self.version)
+        checksum_file = ("%s-%s-%s-CHECKSUM" %
+                         (self.distro, self.version, self.arch))
```

Comment:
I dislike that the ISO and checksum file names are non-uniform. If you're just copying those formats from CentOS, then that's fine.